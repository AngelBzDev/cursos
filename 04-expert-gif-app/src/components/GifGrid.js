import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem'

export const GifGrid = ({ category }) => {

   const { data:images, loading } = useFetchGifs( category );

   

   return (
      <>
         <h3 className="category animate__animated animate__fadeInRight">{ category }</h3>

         { loading && <p>Cargando...</p> }
         
         <div className="card-container animate__animated animate__fadeInLeft">
         { images.map( img  => (
               <GifGridItem 
                  key={ img.id }
                  { ...img }
               />
            ))
         }
         </div>
      </>
   )
}

Type TabNine::sem to enable semantic completion for JavaScript.
This will run the following commands:
Type TabNine::no_sem to suppress this message.
   To learn more about semantic completion, see https://tabnine.com/semantic.


GifGrid.propTypes = {
   category: PropTypes.string.isRequired
}

