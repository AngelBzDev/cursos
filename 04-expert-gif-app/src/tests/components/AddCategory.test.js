import { shallow } from "enzyme";
import {AddCategory} from '../../components/AddCategory';

describe('Pruebas en el componente AddCategory', () => {

   const setCategories = jest.fn();
   let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

   beforeEach(() => {
      jest.clearAllMocks();
      wrapper = shallow(<AddCategory setCategories={setCategories}/>);
   })

   test('Debe de mostrarse corectamente', () => {
      expect(wrapper).toMatchSnapshot();
   })

   test('Debe de cambiar el input', () => {
      const inp = wrapper.find('input'); 
      const value = 'El bicho';
      inp.simulate('change', {target: value});
      
   })

   test('No debe de postear la informacion con submit', () => {
      wrapper.find('form').simulate('submit', { preventDefault(){} });
      expect(setCategories).not.toHaveBeenCalled();
   })

   test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
      const value = 'El bicho';
      const inp = wrapper.find('input');
      
      inp.simulate('change', {target: {value}});
      wrapper.find('form').simulate('submit', { preventDefault(){} });
      expect(setCategories).toHaveBeenCalled();
      expect(setCategories).toHaveBeenCalledTimes(1) //Validar las veces que se llama la funcion
      expect(setCategories).toHaveBeenCalledWith(expect.any(Function)) //Validar el tipo de dato que recibe la funcion
      expect(inp.prop('value')).toBe('');
   })
   
})
