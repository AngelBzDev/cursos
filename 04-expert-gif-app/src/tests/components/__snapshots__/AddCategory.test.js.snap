// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory Debe de mostrarse corectamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Ingrese la categoria"
    type="text"
    value=""
  />
</form>
`;
